### PR TITLE
Adding POST APIs used to Upload customer TDE certificates in CMS.

### PR DESCRIPTION
**Pull request by nivimsft**
Commit contains appropriate tests

## Description
Adding POST APIs used to Upload customer TDE certificates in CMS code and tests
PR for spec changes https://github.com/Azure/azure-rest-api-specs/pull/3248

---

This checklist is used to make sure that common guidelines for a pull request are followed.
- [X] Please add REST spec PR link to the SDK PR
- [X] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [X] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [X] Title of the pull request is clear and informative.
- [X] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [X] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [X] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [X] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code.
- [X] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

**Review by dsgouda — COMMENTED** [review]
Will approve once REST spec is merged and sdk regenerated.

File: src/SDKs/SqlManagement/Management.Sql/Microsoft.Azure.Management.Sql.csproj
Comment (jaredmoo) [inline src/SDKs/SqlManagement/Management.Sql/Microsoft.Azure.Management.Sql.csproj]: I published 1.17 today (on request from vulnerability assessment team), so now you need to bump this up to 1.18 and wipe out the old release notes (i.e. just line 15)

Comment (jaredmoo) [reply src/SDKs/SqlManagement/Management.Sql/Microsoft.Azure.Management.Sql.csproj]: Bump to 1.18 both in this file and in AssemblyInfo.cs

File: src/SDKs/_metadata/sql_resource-manager.txt
Comment (jaredmoo) [inline src/SDKs/_metadata/sql_resource-manager.txt]: You will need to generate from Azure/master before this PR can be merged

File: src/SDKs/SqlManagement/Sql.Tests/UploadTdeCertificateScenarioTests.cs
Comment (jaredmoo) [inline src/SDKs/SqlManagement/Sql.Tests/UploadTdeCertificateScenarioTests.cs]: Tests MUST be fully automated. This means that they must create all prerequisite resources during test setup. If this is not possible, the test must be skipped (see geo restore scenario test for example)

**Review by jaredmoo — CHANGES_REQUESTED** [review]

File: src/SDKs/SqlManagement/Management.Sql/Microsoft.Azure.Management.Sql.csproj
Comment (nivimsft) [reply src/SDKs/SqlManagement/Management.Sql/Microsoft.Azure.Management.Sql.csproj]: Sure will do

File: src/SDKs/_metadata/sql_resource-manager.txt
Comment (nivimsft) [reply src/SDKs/_metadata/sql_resource-manager.txt]: Sounds good! Waiting for PR to be merged

**Review by dsgouda — APPROVED** [review]
LGTM

**Comment by dsgouda** [comment]
Will merge when CIs pass

File: src/SDKs/SqlManagement/Management.Sql/Microsoft.Azure.Management.Sql.csproj
Comment (jaredmoo) [inline src/SDKs/SqlManagement/Management.Sql/Microsoft.Azure.Management.Sql.csproj]: Customers do not know what CMS is. This release note should be more like, "Added support for uploading TDE certificate into servers and managed instances."

**Review by jaredmoo — APPROVED** [review]
Looks good, release note needs a little edit.

File: src/SDKs/SqlManagement/Management.Sql/Microsoft.Azure.Management.Sql.csproj
Comment (nivimsft) [reply src/SDKs/SqlManagement/Management.Sql/Microsoft.Azure.Management.Sql.csproj]: You are right, will address

**Comment by jaredmoo** [comment]
Thanks, now it's perfect!

**Comment by nivimsft** [comment]
I like that, thanks for your help in making it so! :)

**Pull request merged**
